.github/workflows/promote_charm.yaml

```diff
@@ -24,13 +24,21 @@ jobs:
     needs: get-runner-image
     steps:
       - uses: actions/checkout@v3
+      - name: Get build-on value
+        run: |
+          name="ubuntu"
+          channel="20.04"
+          if [ -f charmcraft.yaml ]; then
+            name=$(yq '.bases.[0].build-on.[0].name' charmcraft.yaml)
+            channel=$(yq '.bases.[0].build-on.[0].channel' charmcraft.yaml)
```

Comment:
What happens if there's no match ? 
I'd rather use the alternate value syntax
https://mikefarah.gitbook.io/yq/operators/alternative-default-value

Reply:
Thanks for the suggestion.
I have updated the code with this feature.